### ISSUE TITLE
Improvements to the Advanced Search Page

### ISSUE THREAD

**Issue by BrittanyBunk** [internetarchive/openlibrary#2673]
Related to #541, #2099.

Now that I use the Advanced Search feature more, I realize it's not adequate enough to look for works/editions. Here's my proposal to add:
![image](https://user-images.githubusercontent.com/55302685/69887015-b5f12f00-1299-11ea-81a7-36dd87303c18.png)
I took out some that I felt aren't as important, but it's ok to keep them. I'm hoping these additions will help someone who has some information about a book (but not a lot, and it's not covered in the current Advanced Search layout) to be able to take the multiple bits of limited info that they have to group it together into a search that helps them find what they're looking for. The ones I added in are what I feel would help me out, but if there's something that would help people more, don't feel limited by what I posted - it's ok to add in.

**Comment by jessamynwest** [comment]
Speaking as someone who has interacted with a lot of Advanced Search fields in library catalogs, and would like to see one on OL, I think this could benefit more refinement. If you're offering more options for search I think OL people should think more about search strategy generally and about how people look for more information. 

Right now, as a counterpoint, the Advanced Search on the Internet Archive basically dissuades people from searching because so much of it is inside baseball for IA superfans. Additionally search results pages are super confusing, limiting only sort of works and there are two kinds of searches (i.e. for a known item and for a general idea that are undifferentiated

We want to offer a good searching experience for users and some of these criterion are a lot more important than others (plus, really big search boxes are more friendly to more users). So, within this example

- collapse # of pages and pagination into one (with a max/min setting)
- allow subtitle to be searched within title (no reason not to,most people don't care about the difference)
- what is search by image supposed to do?
- needs to differentiate between controlled vocabulary (i.e. author, language) and not (location, publisher) because you can do fuzzy match with non-controlled vocabulary but controlled vocabulary should drop a user into a list
- is text "full text" here? If not, how does this differ from the current full text search (which right now goes to results from IA and not OL)
- when people are entering multiple values will the result AND them or OR them (this is crucial, users should be able to choose either which also means limiting to fewer fields so that it doesn't just grind away at server resources)

Many if these can be lumped into a really strong/effective "keyword" search which is how library and database catalogs do it. Or an iterative search strategy where you have a user do a search and then limit the results by the facets that are available.

tl;dr OL needs to have a mapped out Search strategy before getting too deep into building an advanced search.

**Comment by BrittanyBunk** [comment]
@jessamynwest Baseball?
- search by image is when you have an image that looks like the cover, but isn't, and you don't know the name of the book.
- Yes, full text. It differentiates, because someone can combine another field with the full text together instead of alone. Like combining text with year would help a lot.
- I don't think we need the iterative (as those are clunky - and I have trouble getting to the info quickly enough. The best is to go back and remove info from the search), just effective keywords.

**Comment by LeadSongDog** [comment]
@BrittanyBunk https://en.wikipedia.org/wiki/Inside_baseball_(metaphor)
Why do you say faceted/filtered/refined search results are "clunky"? They work well at the major catalogues. We should not ignore the fact that they found it worth doing. 

*Hathitrust https://catalog.hathitrust.org/Search/Home?type%5B%5D=all&lookfor%5B%5D=mort%20d%27arthur&page=1&pagesize=20&ft=
*BnF https://catalogue.bnf.fr/rechercher.do?motRecherche=mort+d%27Arthur&critereRecherche=0&depart=0&facetteModifiee=ok
*BL http://explore.bl.uk/primo_library/libweb/action/dlSearch.do?vid=BLVU1&institution=BL&search_scope=LSCOP-ALL&query=any,contains,mort+d%27arthur&tab=local_tab#
*LoC https://catalog.loc.gov/vwebv/search?searchArg=mort+d%27arthur&searchCode=GKEY%5E*&searchType=0&recCount=25&sk=en_US

**Comment by BrittanyBunk** [comment]
@LeadSongDog I have to scroll down and it takes up space/clutters my view. I don't feel they're really necessary, but if it's there, ok. Some websites take too long to load with them too - plus having them on mobile is even harder.

**Comment by BrittanyBunk** [comment]
@jessamynwest alright, now that @LeadSongDog helped me understand, I could answer properly. It's ok to have an advanced search. Maybe for the IA superfans, we could have a 'see more', so it doesn't dissuade most people, but still allows for the superfans. Google does that and I don't see people searching less than before on there, but probably more than ever. So for that, I think it really helps. A lot of people don't use certain search engines because they're lacking (like Ecosia), so I think it's overall beneficial for everyone for it to be really capable.

**Comment by jessamynwest** [comment]
Great -- I am in favor of advanced search in general but it really needs to be part of OL's general strategy because if it doesn't work the way it's supposed it, it will be worse than nothing. Most users aren't advanced searchers. I suggest getting statistics from OL that looks at how many times overall people are using anything but the basic search and what they are looking for, maybe that can help us craft a strategy.

**Comment by LeadSongDog** [comment]
@jessamynwest If it is possible to obtain, stats on sequential basic searches (refine and retry attempts) would also be of interest.

**Comment by BrittanyBunk** [comment]
@jessamynwest that actually wouldn't be accurate, because if there's something not available on the advanced search, then they won't have stats on it. The better way is to look at websites that do have stats on it - like search engines like Google - and see how many people use those. Would that help?

**Comment by tfmorris** [comment]
>  I suggest getting statistics from OL 

Data driven decision making? Love it!

**Comment by jessamynwest** [comment]
> @jessamynwest that actually wouldn't be accurate... The better way is to look at websites that do have stats on it - like search engines like Google - and see how many people use those. Would that help?

I don't think so. Google isn't a library. We need stats from library catalogs because OL has a very specific kind of item which people will look for in a certain way. This is especially true for items with authority control (things like Author names where there is one canonical name in the system in an optimal situation) and Google has nothing like that.

> > I suggest getting statistics from OL
> 
> Data driven decision making? Love it!

I can dream.

**Comment by BrittanyBunk** [comment]
@jessamynwest I'm not stopping you from getting them if that's what you feel will help you. I just don't see how it's going to help me - as it would be misleading for me - so I personally won't - unless you really want me to.
